tests/test_regex.py

```diff
@@ -116,6 +116,16 @@ def test_wildcard(self):
         self.assertTrue(re.issubset('a.b', '...', input_symbols=input_symbols))
         self.assertTrue(re.issuperset('.', 'a|b', input_symbols=input_symbols))
 
+    def test_shuffle(self):
+        """Should correctly check shuffle"""
+
+        input_symbols = {'a', 'b', 'c', 'd'}
+
+        self.assertTrue(re.isequal('a@b', 'ab|ba', input_symbols=input_symbols))
+        self.assertTrue(re.isequal('ab@cd', 'abcd | acbd | cabd | acdb | cadb | cdab', input_symbols=input_symbols))
+        self.assertTrue(re.isequal('(a*)@(b*)@(c*)@(d*)', '.*', input_symbols=input_symbols))
+        self.assertTrue(re.isequal('ca@db', '(c@db)a | (ca@d)b', input_symbols=input_symbols))
```

Comment:
Some extra tests `a@''`, `''@''` (should these two work?), `ab@ab`, `a*@ba`, `a@(b|c)`

Reply:
@EduardoGoulart1 @eliotwrobson I think the proper way to indicate empty string in a regex would be no characters at all between the obvious word boundaries. For example:

1. `a@`
2. `(x@)`

This is consistent with how the `|` operator is handled in PCRE regex (e.g. `a|`):

```
>>> import re
>>> re.search('(a|)', 'a')
<re.Match object; span=(0, 1), match='a'>
>>>  re.search('(a|)', '')
<re.Match object; span=(0, 0), match=''>
```

Reply:
That sounds reasonable, but is a bit out of scope for this PR (requires some kinda technical changes to the parsing algorithm). Though you can indicate `a|` with the question mark optional, as `a?`.

Reply:
@eliotwrobson Oh yes, of course. Then I think `a@''` and  `''@''` cases aren't necessary to have.